x-pack/test/functional/services/transform/discover.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export function TransformDiscoverProvider({ getService }: FtrProviderContext) {
+  const find = getService('find');
+  const testSubjects = getService('testSubjects');
+
+  return {
+    async assertDiscoverQueryHits(expectedDiscoverQueryHits: string) {
+      await testSubjects.existOrFail('discoverQueryHits');
+
+      const actualDiscoverQueryHits = await testSubjects.getVisibleText('discoverQueryHits');
+
+      expect(actualDiscoverQueryHits).to.be(expectedDiscoverQueryHits);
+    },
+
+    async assertNoResults(expectedDestinationIndex: string) {
+      // Discover should use the destination index pattern
+      const actualIndexPatternSwitchLinkText = await (
+        await testSubjects.find('indexPattern-switch-link')
+      ).getVisibleText();
+      expect(actualIndexPatternSwitchLinkText).to.be(expectedDestinationIndex);
```

Comment:
Similar to my comment above: would be good to have an assertion message here.

Reply:
Fixed in 2616008.